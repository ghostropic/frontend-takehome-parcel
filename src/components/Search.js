import React from 'react'
import styled from 'styled-components'
import { Input, Button } from '@material-ui/core'

const StyledWrapper = styled.div`
  height: 60px;
  margin-top: 30px;
  margin-bottom: 30px;
`

export default class Search extends React.Component {
  constructor(props) {
    super(props)
    this.onSubmit = this.onSubmit.bind(this)
    this.onChange = this.onChange.bind(this)
  }
 
  onSubmit (e) {
    e.preventDefault()
    const query = e.target.elements.query.value
    this.props.handleSearchResults(query)
  }

  onChange (e) {
    const query = e.target.value
    this.props.handleInputChange(query)
  }
 
  render() {
    return (
      <StyledWrapper>
        <form onSubmit={this.onSubmit}>
          <Input
            onChange={this.onChange}
            type="text"
            placeholder="Search Ruby gems."
            name="query"
          />
          <Button type="submit">Search</Button>
        </form>
      </StyledWrapper>
    )
  }
}
