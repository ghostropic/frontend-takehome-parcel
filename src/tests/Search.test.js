import React from 'react'
import { shallow } from 'enzyme'
import { Input, Button } from '@material-ui/core'

import Search from '../components/Search'

test('should render Search component correctly', () => {
  const wrapper = shallow(<Search />)
  expect(wrapper).toMatchSnapshot()
})

test('should call handleInputChange prop on submit', () => {
  const handleSearchResultsSpy = jest.fn()
  const wrapper = shallow(<Search handleSearchResults={handleSearchResultsSpy} />)
  wrapper.find('form').simulate('submit', {
    preventDefault: () => { },
    target: { elements: { query: 'value' } }
  })
  expect(handleSearchResultsSpy).toHaveBeenCalled()
})

test('should call handleInputChange on change', () => {
  const value = 'hi'
  const handleInputChangeSpy = jest.fn()
  const wrapper = shallow(<Search handleInputChange={handleInputChangeSpy} />)
  const query = { target: { value } }
  wrapper.find(Input).prop('onChange')(query)
  expect(handleInputChangeSpy).toHaveBeenCalledWith(value)
})