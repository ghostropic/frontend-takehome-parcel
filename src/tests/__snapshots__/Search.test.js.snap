// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Search component correctly 1`] = `
<styled.div>
  <form
    onSubmit={[Function]}
  >
    <WithStyles(Input)
      name="query"
      onChange={[Function]}
      placeholder="Search Ruby gems."
      type="text"
    />
    <WithStyles(Button)
      type="submit"
    >
      Search
    </WithStyles(Button)>
  </form>
</styled.div>
`;
